src/lib/pubkey/frodokem/frodo_constants.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * FrodoKEM modes and constants
+ * Based on the MIT licensed reference implementation by the designers
+ * (https://github.com/microsoft/PQCrypto-LWEKE/tree/master/src)
+ *
+ * The Fellowship of the FrodoKEM:
+ * (C) 2023 Jack Lloyd
+ *     2023 René Meusel, Amos Treiber - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#include <botan/internal/frodo_constants.h>
+
+#include <botan/xof.h>
+
+namespace Botan {
+
+FrodoKEMConstants::FrodoKEMConstants(FrodoKEMMode mode) : m_mode(mode), m_len_a(128), m_n_bar(8) {
+#if !defined(BOTAN_HAS_AES)
+   BOTAN_ARG_CHECK(!mode.is_aes(), "cannot instantiate AES-based FrodoKEM: This build does not support AES");
+#endif
```

Comment:
Instead of having to do a macro check along with `mode.is_aes()` let's add

`mode.is_available()` (or `is_supported` or ...)

and use it here, in `speed.cpp`, and anywhere else we're doing an explicit `BOTAN_HAS_AES` check.

This also generalizes if we decide to make SHAKE modes optional in the future.

Reply:
I added this method and used it when possible, but in some instances (row generation, KATs, and `PK_Key_Generation_Test`) compile time checks were still needed.